MailCore/Models/Contact/CommonContactCache.swift

```diff
@@ -0,0 +1,74 @@
+/*
+ Infomaniak Mail - iOS App
+ Copyright (C) 2022 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import InfomaniakCore
+import InfomaniakDI
+
+/// A standard cache for `CommonContact`, used by DI
+public typealias ContactCache = NSCache<NSNumber, CommonContact>
+
+/// Creating a `CommonContact` is expensive, relying on a cache to reduce hangs
+public enum CommonContactCache {
+    /// The underlying standard cache
+    @LazyInjectService private static var cache: ContactCache
+
+    /// Get a contact from cache if any or nil
+    public static func getContactFromCache(contactConfiguration: ContactConfiguration) -> CommonContact? {
```

Comment:
The public API of `CommonContactCache` is static func for clarity. But I'm open to use DI as well and use instance methods.